sources/packages/backend/apps/workers/src/controllers/cra-integration/cra-integration.controller.ts

```diff
@@ -0,0 +1,75 @@
+import { Controller } from "@nestjs/common";
+import { ZeebeWorker } from "../../zeebe";
+import {
+  ZeebeJob,
+  MustReturnJobActionAcknowledgement,
+  IOutputVariables,
+  ICustomHeaders,
+} from "zeebe-node";
+import { CRAIncomeVerificationService } from "../../services";
+import {
+  CheckIncomeRequestJobInDTO,
+  CheckIncomeRequestJobOutDTO,
+  CreateIncomeRequestJobInDTO,
+  CreateIncomeRequestJobOutDTO,
+} from "..";
+import {
+  APPLICATION_ID,
+  REPORTED_INCOME,
+  SUPPORTING_USER_ID,
+  TAX_YEAR,
+} from "../workflow-variables";
+
+@Controller()
+export class CRAIntegrationController {
+  constructor(
+    private readonly incomeVerificationService: CRAIncomeVerificationService,
+  ) {}
+
+  @ZeebeWorker("create-income-request", {
+    fetchVariable: [
+      APPLICATION_ID,
+      SUPPORTING_USER_ID,
+      TAX_YEAR,
+      REPORTED_INCOME,
+    ],
+  })
+  async createSupportingUsers(
+    job: Readonly<
+      ZeebeJob<
+        CreateIncomeRequestJobInDTO,
+        IOutputVariables,
+        CreateIncomeRequestJobOutDTO
+      >
+    >,
+  ): Promise<MustReturnJobActionAcknowledgement> {
+    const incomeRequest =
+      await this.incomeVerificationService.createIncomeVerification(
+        job.variables.applicationId,
+        job.variables.taxYear,
+        job.variables.reportedIncome,
+        job.variables.supportingUserId,
+      );
+    const [identifier] = incomeRequest.identifiers;
+    return job.complete({ incomeVerificationId: identifier.id });
+  }
+
+  @ZeebeWorker("check-income-request", {
+    fetchVariable: ["incomeVerificationId"],
+  })
+  async checkSupportingUserResponse(
```

Comment:
Method is named checkSupportingUserResponse (checkIncomeVerification or checkIncomeVerificationStatus) could be more helpful.

Reply:
Changed.